src/intervals/interval_operations/numeric.jl

```diff
@@ -84,43 +84,58 @@ Implement the `mid` function of the IEEE Standard 1788-2015 (Table 9.2).
 
 See also: [`inf`](@ref), [`sup`](@ref), [`bounds`](@ref), [`diam`](@ref),
 [`radius`](@ref) and [`midradius`](@ref).
+
+
+    mid(x, α)
+
+Relative midpoint of `x`, for `α` between 0 and 1.
+
+`mid(x, 0)` is the lower bound of the interval, `mid(x, 1)` the upper bound,
+and `mid(x, 0.5)` the midpoint.
 """
-function mid(x::BareInterval{T}) where {T<:AbstractFloat}
+function mid(x::BareInterval{T}, α = 0.5) where {T<:AbstractFloat}
     isempty_interval(x) && return convert(T, NaN)
-    isentire_interval(x) && return zero(T)
-    lo, hi = bounds(x)
-    lo == typemin(T) && return nextfloat(lo) # cf. Section 12.12.8
-    hi == typemax(T) && return prevfloat(hi) # cf. Section 12.12.8
-    midpoint = (lo + hi) / 2
-    isfinite(midpoint) && return _normalisezero(midpoint)
-    # fallback in case of overflow
-    # cannot be the default, since it does not pass several IEEE 1788-2015 tests for small floats
-    return _normalisezero(lo / 2 + hi / 2)
-end
-function mid(x::BareInterval{T}) where {T<:Rational}
+    if isentire_interval(x)
+        α == 0.5 && return zero(T)
+        α > 0.5 && return prevfloat(typemax(T))
+        return nextfloat(typemin(T))
+    else
+        lo, hi = bounds(x)
+        lo == typemin(T) && return nextfloat(lo) # cf. Section 12.12.8
+        hi == typemax(T) && return prevfloat(hi) # cf. Section 12.12.8
+        β = convert(T, α)
+        midpoint = β * (hi + lo * (1/β - 1)) # Exactly 0.5 * (hi + lo) for β = 0.5
```

Comment:
Why not `midpoint = β * (hi - lo) + lo` (which is what is used for the case of rational interval bounds)?

Reply:
For some small floats you get a slightly different results, and the ITF suite doesn't pass. Namely:

```julia
minimal_mid_test: Test Failed at C:\Users\Kolaru\.julia\dev\IntervalArithmetic\test\ITF1788_tests\libieeep1788_num.jl:153
  Expression: mid(bareinterval(-5.0e-324, 1.0e-323)) === 0.0
   Evaluated: 5.0e-324 === 0.0
```

Reply:
I see